### DISCUSSION TITLE
Trim attributes to a certain length

### DISCUSSION THREAD

**Discussion by zilladotexe** [open-telemetry/opentelemetry-collector-contrib#17541] [Q&A]
We're running into an issue with our APM discarding Logs because they exceed length limits (for attribute name and/or value). I've been looking at the documentation for the attribute processor to try and determine if it's possible perform a trim but unless I'm mistaken, there isn't any kind of trim function?

**Comment by fatsheep9146** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/ottlfuncs/README.md#truncate_all

maybe this function is working?
